hat/assets/js/apps/Iaso/domains/entities/components/EntitiesQuerybuilder/index.tsx

```diff
@@ -0,0 +1,231 @@
+import React, {
+    FunctionComponent,
+    useState,
+    useEffect,
+    useCallback,
+    useMemo,
+} from 'react';
+
+import {
+    QueryBuilderFields,
+    useHumanReadableJsonLogic,
+    useSafeIntl,
+} from 'bluesquare-components';
+import { useDynamicFormDescriptors } from 'Iaso/domains/forms/fields/hooks/useGetFormDescriptor';
+import { getQueryBuildersFields } from 'Iaso/domains/forms/fields/hooks/useGetQueryBuildersFields';
+import { useDynamicPossibleFields } from 'Iaso/domains/forms/hooks/useGetPossibleFields';
+import { Popper } from '../../../forms/fields/components/Popper';
+import { useGetQueryBuilderListToReplace } from '../../../forms/fields/hooks/useGetQueryBuilderListToReplace';
+import { parseJson } from '../../../instances/utils/jsonLogicParse';
+import { useGetForms } from '../../entityTypes/hooks/requests/forms';
+import MESSAGES from '../../messages';
+import { DialogBuilder } from './DialogBuilder';
+import {
+    parseInitialLogic,
+    getAllFields,
+    LogicOperator,
+    FormState,
+    defaultFormState,
+} from './utils';
+
+type Props = {
+    fieldsSearchJson: string;
+    handleChange: (key: string, value?: string) => void;
+};
+
+export const EntitiesQueryBuilder: FunctionComponent<Props> = ({
+    fieldsSearchJson,
+    handleChange,
+}) => {
+    const { data: formsList, isFetching: isFetchingForms } = useGetForms(true);
+    const [allFields, setAllFields] = useState<QueryBuilderFields>({});
+    const [not, setNot] = useState(false);
+    const [activeOperator, setActiveOperator] = useState<'and' | 'or' | null>(
+        'and',
+    );
+    const [formStates, setFormStates] = useState<FormState[] | undefined>();
```

Comment:
```suggestion
    const [formStates, setFormStates] = useState<FormState[]>([]);
```

If you need to check `formState` for content, you can always add a const `hasFormState = Boolean(formState.length)`